aopy/analysis.py

```diff
@@ -856,3 +854,128 @@ def linear_fit_analysis2D(xdata, ydata, weights=None, fit_intercept=True):
+            | **accLLR (npts):** AccLLR time series
+            | **selection_time_idx (int):** Time where AccLLR crosses the threshold
+    '''
+    # Ensure data is 2D
+    if len(data.shape) == 1:
+        data = data[:,None]
+    npts = data.shape[0]
+    ntrials = data.shape[1]
+    LLR = np.zeros(npts)*np.nan # LLR at exact time points
+    
+    if modality == 'spikes':
+        # If modality is spikes, ensure data is binary (all 0's and 1's)
+        binary_spike_mask = np.logical_and(~np.isclose(data, 0), ~np.isclose(data,1))
+        if np.sum(binary_spike_mask) > 0:
+            warnings.warn('Input spiking activity is not binary (all 0s and 1s)')
+
+        # Calculate AccLLR across all trials at each point
+        for ipt in range(npts):
+            temp_LLR = np.zeros(ntrials)*np.nan
+            for itrial in range(ntrials):
+                temp_LLR[itrial] = (cond2[ipt] - cond1[ipt])*bin_width + data[ipt, itrial]*np.log(cond1[ipt]/cond2[ipt])
+            
+            LLR[ipt] = np.nansum(temp_LLR)
+        
+    
+    elif modality == 'lfp':
+    
+        # Calculate AccLLR parameters
+        sigma_sq = np.var(data)
+    
+        # Calculate AccLLR across all trials at each point
+        for ipt in range(npts):
+            temp_LLR = np.zeros(ntrials)*np.nan
+            for itrial in range(ntrials):
+            
+                temp_LLR[itrial] = ((data[ipt, itrial]-cond2[ipt])**2) - ((data[ipt, itrial]-cond1[ipt])**2)/2*sigma_sq
+            
+            LLR[ipt] = np.nansum(temp_LLR)
+
+    else:
+        warnings.warn('Plesae input a valid modality')
+    
+    accLLR = np.cumsum(LLR)
+    thresh_val = thresh_proportion*np.max(np.abs(accLLR))
+    selection_time_idx = np.where(np.abs(accLLR)>thresh_val)[0][0]
+    
+    return accLLR, selection_time_idx*bin_width
+
+def calc_accLLR_threshold(data_cond1, data_cond2, modality, bin_width, step_size=0.01, false_alarm_prob=0.05):
+    '''
+    Sweeps the AccLLR method over the thresh_proportion parameter, estimates false alarm rates, 
+    and then choose a value for thresh_proportion that gives us the desired false alarm rate.
+    
+    Estimate the false alarm probability in the accLLR method by the proportion of 
+    trials from condition 2 whose AccLLR hit the upper detection threshold within the maximum 
+    accumulation time. 
+
+    Banerjee A, Dean HL, Pesaran B. A likelihood method for computing selection times in spiking and local field potential activity. J Neurophysiol. 2010 Dec;104(6):3705-20. doi: 10.1152/jn.00036.2010. Epub 2010 Sep 8.
+    https://pubmed.ncbi.nlm.nih.gov/20884767/
+    '''
+
+    thresh_props = np.arange(0, 1, step_size)
+    fa_rates = []
+
+    # Define some test data and some training data?
+    cond1 = data_cond1[:,0]
+    cond2 = data_cond2[:,0]
+    data = np.concatenate((data_cond1[:,1:], data_cond2[:,1:]))
+    data_labels = np.concatenate((np.ones(len(data_cond1)-1), 2*np.ones(len(data_cond2)-1)))
```

Comment:
I think we need to have the user input training and test data for each condition.